internal/controller/controllers/agent_controller.go

```diff
@@ -204,6 +207,113 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	return r.updateStatus(ctx, log, agent, &h.Host, h.ClusterID, nil, false)
 }
 
+func findMatchingNode(agent *aiv1beta1.Agent, nodes []corev1.Node) *corev1.Node {
+	addresses := make([]string, 0)
+	for _, intf := range agent.Status.Inventory.Interfaces {
+		for _, ip := range append(intf.IPV4Addresses, intf.IPV6Addresses...) {
+			splitIp := strings.Split(ip, "/")
+			if len(splitIp) == 2 {
+				addresses = append(addresses, splitIp[0])
+			}
+		}
+	}
+	for i := range nodes {
+		n := &nodes[i]
+		if strings.EqualFold(n.Name, agent.Spec.Hostname) {
+			return n
+		}
+		for _, addr := range n.Status.Addresses {
+			if addr.Type == corev1.NodeInternalIP && funk.ContainsString(addresses, addr.Address) {
+				return n
+			}
+		}
+	}
+	return nil
+}
+
+func isNodeReady(node *corev1.Node) bool {
+	for _, c := range node.Status.Conditions {
+		if c.Status == corev1.ConditionTrue && c.Type == corev1.NodeReady {
+			return true
+		}
+	}
+	return false
+}
+
+func isNodeApproved(agent *aiv1beta1.Agent, nodes []corev1.Node) bool {
+	if agent.Status.DebugInfo.State == models.HostStatusAddedToExistingCluster &&
+		agent.Status.Progress.CurrentStage == models.HostStageRebooting {
+		node := findMatchingNode(agent, nodes)
+		if node != nil {
+			return isNodeReady(node)
+		}
+	}
+	return false
+}
+
+func (r *AgentReconciler) shouldApproveCSRs(agent *aiv1beta1.Agent, clients ClusterKubeClient) (bool, error) {
+
+	nodes, err := clients.ListNodes()
+	if err != nil {
+		return false, err
+	}
+	return !isNodeApproved(agent, nodes.Items), nil
+}
+func isCsrApproved(csr *certificatesv1.CertificateSigningRequest) bool {
+	for _, c := range csr.Status.Conditions {
+		if c.Type == certificatesv1.CertificateApproved {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *AgentReconciler) ApproveAllCsrs(clients ClusterKubeClient) error {
```

Comment:
how does it apply to a specific node? it looks like you are approving all the hosts in the cluster no?

Reply:
Yes.  I did something similar to this https://github.com/openshift/assisted-installer/blob/e3e0586daac9b07b93f5ca2d220ac0c458f441b3/src/assisted_installer_controller/assisted_installer_controller.go#L368-L384
I will have to find out if the CSR can be associated with the specific agent.

Reply:
Seems to me that the association cannot be recognized

Reply:
@flaper87 @tsorya @eranco74 maybe one of you know how to identify the correct agent on the spoke cluster?

Reply:
It can be done if the CSR itself is opened and the node itself is part of the CN field

Reply:
Do we want to go for it?

Reply:
I have commented in the mail thread but, to provide a more specific comment here, I would rather not approve the CSRs manually if we can let the `machine-approver` do it for us. Let's first understand what it would take for us to integrate with the regular workflow.

Reply:
@flaper87 do you think the similar code https://github.com/openshift/assisted-installer/blob/e3e0586daac9b07b93f5ca2d220ac0c458f441b3/src/assisted_installer_controller/assisted_installer_controller.go#L368-L384 in assisted-installer should be removed as well ?

Reply:
If we have machine approver in None platform, i agree that better to create relevant machine yamls as we do on BM platform, it will allow to have almost same code for all the platforms